test/extensions/filters/network/common/fuzz/uber_writefilter.cc

```diff
@@ -0,0 +1,123 @@
+  // Clear the pointers inside the mock_dispatcher
+  Event::MockDispatcher& mock_dispatcher =
+      dynamic_cast<Event::MockDispatcher&>(write_filter_callbacks_->connection_.dispatcher_);
+  mock_dispatcher.to_delete_.clear();
+  write_filter_.reset();
+}
+
+void UberWriteFilterFuzzer::fuzzerSetup() {
+  // Setup process when this fuzzer object is constructed.
+  // For a static fuzzer, this will only be executed once.
+
+  // Get the pointer of write_filter when the write_filter is being added to connection_.
+  write_filter_callbacks_ = std::make_shared<NiceMock<Network::MockWriteFilterCallbacks>>();
+  read_filter_callbacks_ = std::make_shared<NiceMock<Network::MockReadFilterCallbacks>>();
+  ON_CALL(write_filter_callbacks_->connection_, addWriteFilter(_))
+      .WillByDefault(Invoke([&](Network::WriteFilterSharedPtr write_filter) -> void {
+        write_filter->initializeWriteFilterCallbacks(*write_filter_callbacks_);
+        write_filter_ = write_filter;
+      }));
+  ON_CALL(write_filter_callbacks_->connection_, addFilter(_))
+      .WillByDefault(Invoke([&](Network::FilterSharedPtr filter) -> void {
+        filter->initializeReadFilterCallbacks(*read_filter_callbacks_);
+        filter->initializeWriteFilterCallbacks(*write_filter_callbacks_);
+        write_filter_ = filter;
+      }));
+  factory_context_.prepareSimulatedSystemTime();
+
+  // Set featureEnabled for mongo_proxy
+  ON_CALL(factory_context_.runtime_loader_.snapshot_, featureEnabled("mongo.proxy_enabled", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_,
+          featureEnabled("mongo.connection_logging_enabled", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_, featureEnabled("mongo.logging_enabled", 100))
+      .WillByDefault(Return(true));
+
+  // Set featureEnabled for thrift_proxy
+  ON_CALL(factory_context_.runtime_loader_.snapshot_,
+          featureEnabled("ratelimit.thrift_filter_enabled", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_,
+          featureEnabled("ratelimit.thrift_filter_enforcing", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_,
+          featureEnabled("ratelimit.test_key.thrift_filter_enabled", 100))
+      .WillByDefault(Return(true));
+}
+
+UberWriteFilterFuzzer::UberWriteFilterFuzzer()
+    : time_source_(factory_context_.simulatedTimeSystem()) {
+  fuzzerSetup();
+}
+
+void UberWriteFilterFuzzer::fuzz(
+    const envoy::config::listener::v3::Filter& proto_config,
+    const Protobuf::RepeatedPtrField<::test::extensions::filters::network::WriteAction>& actions) {
+  try {
+    // Try to create the filter callback(cb_). Exit early if the config is invalid or violates PGV
+    // constraints.
+    const std::string& filter_name = proto_config.name();
+    ENVOY_LOG_MISC(info, "filter name {}", filter_name);
```

Comment:
suggest: debug level log here and elsewhere?

Reply:
> suggest: debug level log here and elsewhere?

Modified. Thanks!